packages/flutter_tools/lib/src/flx.dart

```diff
@@ -128,7 +132,9 @@ Future<List<String>> assemble({
       .toList();
 
   if (kernelContent != null) {
-    final String platformKernelDill = artifacts.getArtifactPath(Artifact.platformKernelDill);
+    final String platformKernelDill = strongMode ?
+        artifacts.getArtifactPath(Artifact.platformKernelStrongDill) :
```

Comment:
This would require changes in engine's dart_init.cc(line 89) so that correct platform.dill is picked up from the archive.

Reply:
Scratch that. Like you explained, the file will be put into archive under `_kPlatformKernelKey` name.